### PR TITLE
Make usernamespaces work without sidecar process

### PR DESCRIPTION
**Pull request by crosbymichael**
VICTORY

After spending alot of time reading long strace's I finally found a way to get user namespaces with work with all of the standard code.

The big changes is trying to mknod a device, if you get an EPERM then bind mount it, done.

ping @mrunalp !!!!!

### CODE REVIEW

**Comment by mrunalp** [comment]
Great, I got as far as figuring out that we needed to bind mount :)

File: linux_rootfs.go [hunk 1]
```diff
@@ -209,6 +206,28 @@ func createDeviceNode(rootfs string, node *configs.Device) error {
 	if err := os.MkdirAll(parent, 0755); err != nil {
 		return err
 	}
+	if err := mknodDevice(dest, node); err != nil {
+		if os.IsExist(err) {
+			return nil
+		}
+		// containers running in a user namespace are not allowed to mknod
+		// devices so we can just bind mount it from the host.
```
Comment (mrunalp) [inline linux_rootfs.go]: could we pass a host bind flag here instead of relying on EPERM?

**Comment by avagin** [comment]
Good job! Except to tiny comments, LGTM. Thanks.

**Comment by crosbymichael** [comment]
@avagin updated

**Comment by avagin** [comment]
LGTM, thanks

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**